reverse_proxy.go

```diff
@@ -371,6 +371,25 @@ func (p *ReverseProxy) CheckHardTimeoutEnforced(spec *APISpec, req *http.Request
 	return false, globalConf.ProxyDefaultTimeout
 }
 
+func (p *ReverseProxy) CheckHeaderAllowed(hdr string, spec *APISpec, req *http.Request) bool {
+	log.Debug("Checking if header allowed: ", hdr)
+	vInfo, versionPaths, _, _ := spec.Version(req)
+	deleteHeaders := vInfo.GlobalHeadersRemove
+
+	if found, meta := spec.CheckSpecMatchesStatus(req, versionPaths, HeaderInjected); found {
+		hmeta := meta.(*apidef.HeaderInjectionMeta)
+		deleteHeaders = append(deleteHeaders, hmeta.DeleteHeaders...)
```

Comment:
I would prefer having 2 `for` cycles for both global and request specific headers, to avoid additional memory allocations.